plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/Story.tsx

```diff
@@ -0,0 +1,210 @@
+import AddButton from "@web/extensions/sidebar/core/components/content/common/DatasetCard/AddButton";
+import { array_move, generateID } from "@web/extensions/sidebar/utils";
+import { Icon } from "@web/sharedComponents";
+import { styled } from "@web/theme";
+import { useCallback, useState } from "react";
+
+import { BaseFieldProps, StoryItem } from "../types";
+
+const Story: React.FC<BaseFieldProps<"story">> = ({ value, editMode, onUpdate }) => {
+  const [, updateState] = useState<any>();
+  const forceUpdate = useCallback(() => updateState({}), []);
```

Comment:
Why is this needed? As far as I can tell, this component should act like any other field component, which doesn't need a forceUpdate function like this being called like this. Why is this different?

Reply:
For referenced values like Object and Array  changing their content doesn't flag them as different. It should be a different memory reference.  that's why I tried to force re-render

Reply:
React should re-render if a prop has changed (doesn't matter if object, array, string, etc). If it wasn't re-rendering, please assign value to useState and then use the useState value inside the component locally rather than forcing an update